### ISSUE TITLE
Crash when presenting Camera via Web View in iOS 18.2 Beta - WebCore::AVVideoCaptureSource::create

### ISSUE THREAD

**Issue by pavel-uvarov-dd** [swiftlang/swift#77680]
### Description

We are experiencing thousands of crashes in our application when attempting to present the camera through a Web View. The app crashes during this process, and the crash logs point to

- WebCore::AVVideoCaptureSource::create
- WebCore::RealtimeMediaSourceCenter::getUserMediaDevices.
This issue has **only** been observed **in iOS 18.2 beta** versions (beta 1 - 22C5109p, beta 2 - 22C5125e, beta 3 - 22C5131e).

In iOS versions below 18.2, the functionality works and we haven't identified any correlation with specific device models. The problem seems to stem from a WebCore framework introduced in these beta releases 18.2.

We kindly request a review and fix for this issue in upcoming beta releases to restore functionality. Let us know if there are any workarounds or adjustments we can implement in the interim.

Thank you for your attention to this matter.

### Reproduction

```swift

```

### Stack dump

```text
Crashed in non-app
WebCore
+0x05f3284
WebCore::AVVideoCaptureSource::create
WebCore
+0x2460dd4
WebCore::RealtimeMediaSourceCenter::getUserMediaDevices
WebCore
+0x24614f4
WebCore::RealtimeMediaSourceCenter::validateRequestConstraintsAfterEnumeration
WebCore
+0x245fbe4
WebCore::RealtimeMediaSourceCenter::enumerateDevices
WebCore
+0x2461248
WebCore::RealtimeMediaSourceCenter::validateRequestConstraints
WebKit
+0x0823090
WTF::Detail::CallableWrapper<T>::call
WebKit
+0x07f06ec
WebKit::DeviceIdHashSaltStorage::completeDeviceIdHashSaltForOriginCall
WebKit
+0x07f0858
WebKit::DeviceIdHashSaltStorage::deviceIdHashSaltForOrigin
WebKit
+0x0822c20
WTF::Detail::CallableWrapper<T>::call
WebKit
+0x0823d70
WTF::Detail::CallableWrapper<T>::call
WebKit
+0x08179e0
WebKit::UserMediaPermissionCheckProxy::complete
WebKit
+0x071b398
WebKit::UIDelegate::UIClient::checkUserMediaPermissionForOrigin
WebKit
+0x08191c4
WebKit::UserMediaPermissionRequestManagerProxy::getUserMediaPermissionInfo
WebKit
+0x081b6f8
WebKit::UserMediaPermissionRequestManagerProxy::startProcessingUserMediaPermissionRequest
WebKit
+0x081b15c
WebKit::UserMediaPermissionRequestManagerProxy::requestUserMediaPermissionForFrame
WebKit
+0x0886d28
WebKit::WebPageProxy::requestUserMediaPermissionForFrame
WebKit
+0x00e9504
WebKit::WebPageProxy::didReceiveMessage
WebKit
+0x00e3638
IPC::MessageReceiverMap::dispatchMessage
WebKit
+0x0023948
WebKit::WebProcessProxy::didReceiveMessage
WebKit
+0x0e85cdc
IPC::Connection::dispatchMessage
WebKit
+0x00334d8
IPC::Connection::dispatchIncomingMessages
JavaScriptCore
+0x00590b4
WTF::RunLoop::performWork
JavaScriptCore
+0x0059fdc
WTF::RunLoop::performWork
CoreFoundation
+0x0056324
__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
CoreFoundation
+0x00562b8
__CFRunLoopDoSource0
CoreFoundation
+0x0053dbc
__CFRunLoopDoSources0
CoreFoundation
+0x0052fb8
__CFRunLoopRun
CoreFoundation
+0x005282c
CFRunLoopRunSpecific
GraphicsServices
+0x00011c0
GSEventRunModal
UIKitCore
+0x03d4114
-[UIApplication _run]
UIKitCore
+0x04828ec
UIApplicationMain
UIKitCore
+0x07cf104
block_destroy_helper.23
DasherApp
+0x00a21dc
$sSo21UIApplicationDelegateP5UIKitE4mainyyFZ9DasherApp0fB0C_Tgm5 (AppDelegate.swift)
In App
DasherApp
+0x00a21dc
AppDelegate.$main (<compiler-generated>:16)
In App
DasherApp
+0x00a21dc
main (AppDelegate.swift)
In App
```

### Expected behavior

When we load a web view and the user grants permission to access the camera, we proceed to present the camera successfully without the app crashing.

### Environment

iOS 18.2 beta versions (beta 1 - 22C5109p, beta 2 - 22C5125e, beta 3 - 22C5131e)

**Comment by pavel-uvarov-dd** [comment]
Hi @anushshenoy,

We also reported this issue on WebKit.org [here](https://bugs.webkit.org/show_bug.cgi?id=283339), and it has been addressed by this [PR](https://github.com/WebKit/WebKit/pull/36860). However, we’re unable to confirm the fix as it has not yet been included in the 18.2 beta 4 release.

cc: @jmzdoordash

**Comment by Ajaya1000** [comment]
@pavel-uvarov-dd 
With the above issue another similar issue we are facing for CoreAudioSharedUnit while opening camera with mic.

Crashed: com.apple.main-thread
WebCore::CoreAudioSharedUnit::CoreAudioSharedUnit() 
WebCore::CoreAudioCaptureSource::CoreAudioCaptureSource

Do not understand if they are related. Though the crashes have declined after 20th/21st Nov, wanna make sure this issue is because of WebCore library.
